pkg/admission/validator/lakom/shoot.go

```diff
@@ -44,16 +50,78 @@ func findExtension(extensions []core.Extension, extensionType string) (int, core
 
 func (s *shoot) validateScopeType(fldPath *field.Path, scopeType lakom.ScopeType) field.ErrorList {
 	errList := field.ErrorList{}
-
 	if !lakom.AllowedScopes.Has(scopeType) {
 		errList = append(errList, field.NotSupported(fldPath, scopeType, lakom.AllowedScopes.UnsortedList()))
 	}
 
 	return errList
 }
 
+// TODO: This check exists in the validation of the lakom config as well. It can be extracted as a util function
```

Comment:
Will you address this TODO with this PR or not?

Reply:
So the original code that does these checks works a bit differently with the way it handles the errors:
https://github.com/gardener/gardener-extension-shoot-lakom-service/blob/9ee7121684091d71148300c087a2341c929cb111/pkg/lakom/config/config.go#L42-L57

I would like to extract the common logic but I'm not sure how to reconcile this difference between them.
Maybe the new function could also return some sort of aggregate of all the errors and return it? What do you think?

Reply:
The common function can return `field.ErrorList`, then
- `github.com/gardener/gardener-extension-shoot-lakom-service/pkg/admission/validator/lakom.validateCosignPublicKeys` will return directly the result returned from the common function
- `github.com/gardener/gardener-extension-shoot-lakom-service/pkg/lakom/config.validate` will call `ToAggregate()` on the result returned from the common function and return it.

Reply:
Me and @vpnachev agreed to leave it as is. Resolving